include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SENSS_H_
+#define ZEPHYR_INCLUDE_SENSS_H_
+
+/**
+ * @defgroup senss Sensor Subsystem APIs
+ * @defgroup senss_api API
+ * @ingroup senss
+ * @defgroup senss_sensor_types  Sensor Types
+ * @ingroup senss
+ * @defgroup senss_datatypes Data Types
+ * @ingroup senss
+ */
+
+#include <zephyr/senss/senss_datatypes.h>
+#include <zephyr/senss/senss_sensor_types.h>
+
+/**
+ * @brief Sensor Subsystem API
+ * @addtogroup senss_api
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @struct senss_sensor_version
+ * @brief Sensor Version
+ */
+struct senss_sensor_version {
+	union {
+		uint32_t value;
+		struct {
+			uint8_t major;
+			uint8_t minor;
+			uint8_t hotfix;
+			uint8_t build;
+		};
+	};
+};
+
+#define SENSS_SENSOR_INVALID_HANDLE (-1)
+#define SENSS_INDEX_ALL (-1)
+
+#define SENSS_SENSOR_VERSION(_major, _minor, _hotfix, _build)           \
+	((_major) << 24 | (_minor) << 16 | (_hotfix) << 8 | (_build))
+
+
+/**
+ * @brief Sensor flag indicating if this sensor is on event reporting data.
+ *
+ * Reporting sensor data when the sensor event occurs, such as a motion detect sensor reporting
+ * a motion or motionless detected event.
+ */
+#define SENSS_SENSOR_FLAG_REPORT_ON_EVENT			BIT(0)
+
+/**
+ * @brief Sensor flag indicating if this sensor is on change reporting data.
+ *
+ * Reporting sensor data when the sensor data changes.
+ *
+ * Exclusive with  \ref SENSS_SENSOR_FLAG_REPORT_ON_EVENT
+ */
+#define SENSS_SENSOR_FLAG_REPORT_ON_CHANGE			BIT(1)
+
+
+/**
+ * @brief Sensor state.
+ *
+ */
+enum senss_sensor_state {
+	SENSS_SENOSR_STATE_NOT_READY = 1,
+	SENSS_SENOSR_STATE_READY = 2,
+};
+
+
+/**
+ * @struct senss_sensor_info
+ * @brief Sensor basic constant information
+ *
+ */
+struct senss_sensor_info {
+	/** Name of the sensor instance */
+	const char *name;
+
+	/** Friendly name of the sensor instance */
+	const char *friendly_name;
+
+	/** Vendor name of the sensor instance */
+	const char *vendor;
+
+	/** Model name of the sensor instance */
+	const char *model;
+
+	/** Sensor type */
+	int32_t type;
+
+	/** Sensor index for multiple sensors under same sensor type */
```

Comment:
Replace with "Sensor index in the sensor info array returned from ssense_get_sensors()"

Reply:
modified the description, thanks.